src/queries/label-text.js

```diff
@@ -51,15 +36,38 @@ function queryAllLabelsByText(
     .map(({node}) => node)
 }
 
-function getLabelContent(label) {
-  let labelContent = label.getAttribute('value') || label.textContent
-  Array.from(label.querySelectorAll('textarea')).forEach(textarea => {
-    labelContent = labelContent.replace(textarea.value, '')
-  })
-  Array.from(label.querySelectorAll('select')).forEach(select => {
-    labelContent = labelContent.replace(select.textContent, '')
-  })
-  return labelContent
+function getTextContent(node) {
+  const TEXT_NODE = 3
+  const labelledNodeNames = [
+    'button',
+    'meter',
+    'output',
+    'progress',
+    'select',
+    'textarea',
+    'input',
+  ]
+  if (
+    labelledNodeNames.some(nodeName => node.nodeName.toLowerCase() === nodeName)
+  ) {
+    return ''
+  }
+
+  if (node.nodeType === TEXT_NODE) return node.textContent
+
+  return Array.from(node.childNodes)
+    .map(childNode => getTextContent(childNode))
+    .join('')
+}
+
+function getLabelContent(node) {
+  let textContent
+  if (node.tagName.toLowerCase() === 'label') {
+    textContent = getTextContent(node)
+  } else {
```

Comment:
A question jumps in my mind (I beg you pardon if it is a stupid one): is the desired component that only the `label` elements go through `getTextContent` to obtain their value? Will never happen that a labelling element (not a `label` one) needs to be processed by `getTextContent`?

Reply:
I don't think because the else case should be covered only using the  `aria-labelledby`. in that case we should take everything

Reply:
Got it! Thank you!